tests/sentry/integrations/github/test_integration.py

```diff
@@ -738,3 +738,109 @@ def test_get_trees_for_org_rate_limit_401(self):
                 ("baz", "master", []),
             ]
         )
+
+    @responses.activate
+    def test_get_trees_for_org_makes_API_requests_before_MAX_CONNECTION_ERRORS_is_hit(self):
```

Comment:
![image](https://user-images.githubusercontent.com/14812505/227834332-610422ad-b978-4db1-bd0f-0d26512404a8.png)

Seriously, though, if we insist on the snake case custom being an absolute, inviolable rule, it means we can never refer to any constant or class by name in the test title, which feels like a silly restriction to put on ourselves just for the sake of following a convention. Besides, there are plenty of places [we already do this](https://github.com/search?q=org%3Agetsentry+%2F%28%3F-i%29def+test_.*%5BA-Z%5D.*%5C%28%2F&type=code) (even some [written by Armin](https://github.com/getsentry/raven-python/blame/5b3f48c66269993a0202cfc988750e5fe66e0c00/tests/contrib/sanic/tests.py#L188), who feels like about as good a Python role model as they come).

Can you help me understand why you think strict adherence is so important? In this case it's easy enough to reword the test title, but this has come up in multiple PRs now, and I feel like we should just come to an agreement, even if it's to agree to disagree, so that progress isn't held up in the future.

Reply:
Let's say it's okay in cases where we want to keep resemblance to the original form of some proper name and only for tests. Works for you?

For me, the consistency helps the mind to have expectations on what to encounter rather than having to stop and see why this or that is now slightly difference than the norm, thus, reducing the cognitive load.

I love the meme :)

Reply:
> Let's say it's okay in cases where we want to keep resemblance to the original form of some proper name and only for tests. Works for you?

Yeah, that's all I'm proposing. Random variables/functions/whatever, I agree, should be in snake case because #python. But tests (and maybe mocks) are pretty much unique in that they're metacode, if you will. (They're code which references other code (names) _as code_ rather than as values those names might point to.) They break the fourth wall. So I do think that for the sake of communication, calling things by their names (even if those names happen to involve capital letters) is a valid thing to do.

>For me, the consistency helps the mind to have expectations on what to encounter rather than having to stop and see why this or that is now slightly difference than the norm, thus, reducing the cognitive load.

For what it's worth, I actually generally am stupidly anal about things being consistent, for exactly the reason you name (well, also because it pleases my latent OCD). So I suspect we're actually pretty much on the same page here. 🙂

Reply:
lol we're in good company.